sceptre/diffing/stack_differ.py

```diff
@@ -167,6 +167,12 @@ def _extract_parameters_from_generated_stack(self, stack: Stack) -> dict:
         """
         formatted_parameters = {}
         for key, value in stack.parameters.items():
+            # When boto3 receives "None" for a cloudformation parameter, it treats it as if the
+            # value is not passed at all. To be consistent in our diffing, we need to skip Nones
+            # altogether.
+            if value is None:
+                continue
+
             if isinstance(value, list):
                 value = ",".join(item.rstrip("\n") for item in value)
             formatted_parameters[key] = value.rstrip("\n")
```

Comment:
Do you think we need some sort of proper error message if the value isn't a list of strings, a string, or `None`? I mean, isn't that what you ran into in the first place? That the messaging wasn't clear about what the problem was?

Reply:
@jfalkenstein didn't we already discuss this idea and conclude that this code path would not be reachable in any other scenario since Sceptre would not have allowed us to create the stack in the first place, e.g. if the value is a dict

```
"Parameter validation failed:\nInvalid type for parameter Parameters[0].ParameterValue, value: {'bar': 'baz'}, type: <class 'dict'>, valid types: <class 'str'>"
```

Reply:
Yeah, that's probably fine then.